R/error-checks.R

```diff
@@ -486,3 +500,71 @@ validateHasUniqueValues <- function(data, dataName = "dataset", na.rm = TRUE, nu
   }
   stop(messages$errorHasNoUniqueValues(data, dataName, na.rm))
 }
+
+validateIsIncludedInDataset <- function(columnNames, dataset, datasetName = NULL, nullAllowed = FALSE, logFolder = NULL) {
+  if (nullAllowed && is.null(columnNames)) {
+    return()
+  }
+  if (isIncluded(columnNames, names(dataset))) {
+    return()
+  }
+  if (is.null(logFolder)) {
+    stop(messages$errorNotIncludedInDataset(columnNames, dataset, datasetName), call. = FALSE, immediate. = TRUE)
+  }
+  logErrorThenStop(messages$errorNotIncludedInDataset(columnNames, dataset, datasetName))
+}
+
+checkIsIncludedInDataset <- function(columnNames, dataset, datasetName = NULL, nullAllowed = FALSE, logFolder = NULL) {
+  if (nullAllowed && is.null(columnNames)) {
+    return()
+  }
+  if (isIncluded(columnNames, names(dataset))) {
+    return()
+  }
+  if (is.null(logFolder)) {
+    warning(messages$errorNotIncludedInDataset(columnNames, dataset, datasetName), call. = FALSE, immediate. = TRUE)
+    return()
+  }
+  logWorkflow(
+    message = messages$errorNotIncludedInDataset(columnNames, dataset, datasetName),
+    pathFolder = logFolder,
+    logTypes = c(LogTypes$Debug, LogTypes$Error)
+  )
+}
+
+validateUnitDataDefinition <- function(unit, unitColumn, observedDataset, outputs = NULL) {
+  # In case, value from reading from Excel/csv file is not an actual NULL
+  if (any(isOfLength(unit, 0), is.na(unit), unit %in% "")) {
+    unit <- NULL
+  }
+  # Case unit is defined using outputs
+  dataUnit <- NULL
+  if (!isOfLength(outputs, 0)) {
+    dataUnit <- unlist(lapply(outputs, function(output) {
+      output$dataUnit
+    }))
+  }
+
+  # Checks for errors
+  # If no unit defined at all
+  if (isOfLength(c(unit, unitColumn, dataUnit), 0)) {
+    stop(messages$errorNoDataUnit())
+  }
+  # If no unit defined by dictionray, all outputs need to define dataUnit
+  if (isOfLength(c(unit, unitColumn), 0)) {
+    if (!isSameLength(dataUnit, outputs)) {
+      stop(messages$errorNoDataUnitInOutputs())

```

Comment:
why stop and not logandstop? what's the difference?

Reply:
The error checking of unit is performed when creating a `SimulationSet` object prior the creation of a `Workflow` object.
At that point the user hasn't defined the folder where to save logs.
The question is do we save these errors in the current folder (possibly duplicating the log files) or do we just print the errors on the console ?

Reply:
I see. Sounds good